src/workflow/RunTranslator.cpp

```diff
@@ -38,6 +38,8 @@ void OSWorkflow::runTranslator() {
   auto runDir = workflowJSON.absoluteRunDir();
   OS_ASSERT(openstudio::filesystem::is_directory(runDir));
 
+  applyWeatherFileFromModel();
```

Comment:
Workflow gem seem to be doing it in APplyMeasure when it's a ModelMeasure: https://github.com/NREL/OpenStudio-workflow-gem/blob/696306b5b24cd597edc6ab04faa1e618f99db190/lib/openstudio/workflow/util/measure.rb#L551-L571

Reply:
I believe you actually need to complete this unfinished block instead: https://github.com/NREL/OpenStudio/blob/develop/src/workflow/ApplyMeasure.cpp#L215-L224

Reply:
Right. I forgot about Runner::lastEpwFile. But I think it still needs to happen before translation too, because don't we need to account for the fact that the last ModelMeasure could have changed the weather location?

Reply:
Updating last runner epwfile is done prior to measure run currently. The unfinished code block I link to is supposed to be executed after the measure run.
I'm not against changing what workflow-gem does, but it's 1) easier to Match and 2) clearer and probably 3) less error prone